spec/lib/msf/core/exploit/remote/remote/http/http_cookie_jar_spec.rb

```diff
@@ -0,0 +1,261 @@
+        #empty cookie_jar made in before
+
+        @cookie_jar.clear
+
+        expect(@cookie_jar.empty?).to eql(true)
+      end
+
+      it "will return false when a cookie has been added to a cookie_jar" do
+        c = cookie
+
+        @cookie_jar.add(c)
+        @cookie_jar.clear
+
+        expect(@cookie_jar.empty?).to eql(true)
+      end
+    end
+
+    describe "cookies" do
+      it "will return an empty array when no cookies have been added to the jar" do
+        #cookie_jar made in before
+
+        c_array = @cookie_jar.cookies
+
+        expect(c_array.is_a?(Array))
+        expect(c_array.empty?)
+      end
+
+      it "will return an array of all cookies added to the cookie_jar when called with no url param" do
+        c_array = []
+        rand(1..10).times do
+          c = cookie
+
+          c_array.append(c)
+          @cookie_jar.add(c)
+        end
+
+        jar_array = @cookie_jar.cookies
+
+        expect(c_array.sort == jar_array.sort)
+      end
+
+      it "will return an array of all cookies matching the url passed" do
+        c_array = [cookie, cookie, cookie, cookie]
+        c_array[1].domain = c_array[0].domain
+        c_array[1].path = "#{c_array[0].path}/#{random_string}"
+        c_array.map { |c| @cookie_jar.add(c) }
+
+        require 'pry'
+        binding.pry
+
+        filtered_cookies = @cookie_jar.cookies("http://#{c_array[0].domain}#{c_array[0].path}")
+
+        expect(filtered_cookies.length).to eql(2)
+        expect(filtered_cookies.include?(c_array[0])).to eql(true)
+        expect(filtered_cookies.include?(c_array[1])).to eql(true)
+        expect(filtered_cookies.include?(c_array[2])).to eql(false)
+        expect(filtered_cookies.include?(c_array[3])).to eql(false)
+      end
+    end
+
+    describe "add" do
+      it "unacceptable cookie missing path throws ArgumentError" do
+        c = HttpCookie.new(random_string, random_string,
+                 :path => '/' + random_string
+        )
+
+        expect {
+          @cookie_jar.add(c)
+        }.to raise_error(ArgumentError)
+      end
+
+      it "unacceptable cookie missing domain throws ArgumentError" do
+        c = HttpCookie.new(random_string, random_string,
+                           :domain => random_string
+        )
+
+        expect {
+          @cookie_jar.add(c)
+        }.to raise_error(ArgumentError)
+      end
+
+      it "acceptable cookie added to cookie_jar successfully" do
+        c = cookie
+
+        @cookie_jar.add(c)
+
+        expect(@cookie_jar.cookies[0] == c)
+      end
+
+      it "acceptable cookie added to cookie_jar containing cookie with the same name, domain, and path will result in an overwrite" do
+        c = cookie
+        c_dup = cookie
+        c_dup.name = c.name
+        c_dup.domain = c.domain
+        c_dup.path = c.path
+
+        @cookie_jar.add(c)
+        @cookie_jar.add(c_dup)
+
+        expect(@cookie_jar.cookies[0] == c_dup)
```

Comment:
I don't think this is testing anything, looks like you're missing asserts throughout the file? 👀 

```
expect(@cookie_jar.cookies[0]).to eql(c_dup)
```